packages/connect/src/Detector.ts

```diff
@@ -100,6 +100,24 @@ export class Detector {
    * {@link https://polkadot.js.org/docs/}
    */
   public connect = async (chainName: string, chainSpec?: string, options?: ApiOptions): Promise<ApiPromise> => {
+    const provider: ExtensionProvider | SmoldotProvider = this.provider(chainName, chainSpec);
+    provider.connect().catch(console.error);
+
+    this.#providers[chainName] = provider as ProviderInterface;
+    return await ApiPromise.create(Object.assign(options ?? {}, {provider}));
+  }
+
+  /**
+   * creates and returns a provider (either Smoldot or Extension one) to be used
+   * in PolkadotJS API instance
+   * 
+   * @param chainName - the name of the blockchain network to connect to
+   * @param chainSpec - an optional chainSpec to connect to a different network
+   * @returns a provider will be used in a ApiPromise create for PolkadotJS API
+   *
```

Comment:
Add a remark about not being supported too. Something like

```
@remarks

This is used internally for advanced PolkadotJS use cases and is not supported.  Use {@link connect} instead.
```

Reply:
> Some minor nits on the docs in inline comments and you also need to update the landing page example.
> 
> This should PR should include updates to the demos to be 100% it all works. The @substrate/connect API and it's usage in the demos should always change together in the same PR.

Can you please elaborate what you meam by "updates to the demos"? 
Demos are using afaik `.connect()` and this change should not affect or alter their code

Reply:
ignore me I'm going mad! for some reason I'd got it my head we'd changed this